lib/Wikia/src/Service/User/Preferences/PreferenceModule.php

```diff
@@ -27,8 +30,22 @@ public function configure( InjectorBuilder $builder ) {
 				global $wgHiddenPrefs;
 				return $wgHiddenPrefs;
 			} )
-			->bind( PreferenceServiceImpl::DEFAULT_PREFERENCES )->to( function() {
-				return User::getDefaultPreferences();
+			->bind( PreferenceServiceImpl::DEFAULT_PREFERENCES )->to( function( ContainerInterface $c ) {
+				/** @var PreferenceScopeService $scopeService */
+				$scopeService = $c->get( PreferenceScopeService::class );
+				$defaultOptions = User::getDefaultOptions();
+				$defaultPreferences = new UserPreferences();
+
+				foreach ( $defaultOptions as $name => $val ) {
+					if ( $scopeService->isGlobalPreference( $name ) ) {
+						$defaultPreferences->setGlobalPreference( $name, $val );
+					} elseif ( $scopeService->isLocalPreference( $name ) ) {
+						list( $prefName, $wikiId ) = $scopeService->splitLocalPreference( $name );
+						$defaultPreferences->setLocalPreference( $prefName, $wikiId, $val );
+					}
+				}
+
+				return $defaultPreferences;
```

Comment:
It doesn't seem like this should be happening here. This isn't directly related to DI.